src/coord/src/coord.rs

```diff
@@ -327,9 +328,11 @@ pub struct Coordinator {
     /// Tracks write frontiers for active exactly-once sinks.
     sink_writes: HashMap<GlobalId, SinkWrites<Timestamp>>,
 
-    /// A map from pending peeks to the queue into which responses are sent, and
-    /// the IDs of workers who have responded.
-    pending_peeks: HashMap<u32, mpsc::UnboundedSender<PeekResponse>>,
+    /// A map from pending peek ids to the queue into which responses are sent, and
+    /// the connection id of the client that initiated the peek.
+    pending_peeks: HashMap<Uuid, (mpsc::UnboundedSender<PeekResponse>, u32)>,
+    /// A map from client connection ids to a set of all pending peeks for that client
+    client_pending_peeks: HashMap<u32, HashSet<Uuid>>,
```

Comment:
If I'm understanding correctly, it seems like it's not *currently* possible for a single connection to have multiple in-flight peeks. Meaning that this `HashSet` will never get bigger than a size of 1. However we are likely in the future to add some functionality that would allow a single connection to have multiple in-flight peeks.

I decided to make it this a `HashSet` to help support that future use case. Similarly I switched  the `CancelPeeks` command to be able to cancel multiple peeks at once. If my understanding is not correct, or if this is premature, then I can switch the implementation to only expect a single pending peek per connection.

Reply:
> However we are likely in the future to add some functionality that would allow a single connection to have multiple in-flight peeks.

We are? Portals or something?

Come to think of it, do portals already allow for this? Not sure.

Reply:
Well, I'm just trying to interpret the issue description. Though I'm not sure that I'm doing that correctly.
>  For example, active replicas load up a sequence of commands, which may include many peeks with the same identifier, and corresponding responses all with the same identifier.

Reply:
Ah, I see! Hm. Not sure! I'm not sure why we would want COMPUTE to replay peeks, but I am probably missing something. Using a `HashSet` seems like a totally fine call for now! We can always ditch it later.

Reply:
Ah, just for context: the known problems are less with the coordinator firing off multiple peeks concurrently, and more with a replica spinning up and the Compute layer presenting the command history at it. And then having to disambiguate its responses. Ideally all without doing the peeks in sequence, so that only one is active at a time. Part of the confusion is that we don't cancel satisfied peeks, so it can be a bit tricky to suss out which peeks the coordinator is still waiting for.

Reply:
> Come to think of it, do portals already allow for this? Not sure.

My read of the code is that portals would allow multiple pending peeks for a single connection. I doesn't seem to be tested though.